pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -1016,22 +1017,39 @@ private int numReplicasToLookFor() {
       return hold(instanceId, offset);
     }
 
-    // Pick a winner, preferring this instance if tied for highest.
-    // Side-effect: Sets the _winner and _winningOffset
-    private void pickWinner(String preferredInstance) {
-      long maxSoFar = -1;
-      String winner = null;
-      for (Map.Entry<String, Long> entry : _commitStateMap.entrySet()) {
-        if (entry.getValue() > maxSoFar) {
-          maxSoFar = entry.getValue();
-          winner = entry.getKey();
+    /**
+     * Pick a winner if we can, preferring the instance that we are handling right now,
+     *
+     * @param preferredInstance The instance that is reporting in this thread.
+     * @param now current time
+     * @param stopReason reason reported by instance for stopping consumption.
+     * @return true if winner picked, false otherwise.
+     */
+    private boolean pickedWinner(String preferredInstance, long now, final String stopReason) {
```

Comment:
The reason I changed it to pickedWinner was that the return of this method is a yes/no answer depending on whether  a winner was picked or not. I can change it to didPickWinner() or winnerPicked() or isWinnerPicked()